tests/qa_automation/execute_test_run.pm

```diff
@@ -0,0 +1,56 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary:  to execute testsuites in openQA from QA:Head by running its build up test run script
+# Maintainer: Yong Sun  <yosun@suse.com>
+
+use strict;
+use File::Basename;
+use IO::File;
+use Data::Dumper;
+use utils;
+use testapi;
+use ctcs2_to_junit;
+use upload_system_log;
+use base "opensusebasetest";
+
+sub run {
+    my ($self, %args) = @_;
+    my $timeout = $args{timeout} || 180 * 60;
```

Comment:
Setting default timeout to 10800 (180*60) could be too high, as the default whole timeout is lesser than that (from what I remember it's 7200).
Please consider using MAX_JOB_TIME in your test configuration instead of setting a high timeout in the code, or be sure to always set the correct timeout when your test is called.

Reply:
default timeout not enough for especially stress testing. 180 * 60 has verified in the current

Reply:
Ok, but how 180*60 can be reached it you don't use MAX_JOB_TIME in your test configuration? As by default the global timeout is 7200 (I just verified it), so lesser than your default timeout.

Will see what another reviewer with merge access will say :-)

Reply:
seems that `%args` could never be used in the case. and i can see your prove run completed in around 1 hour.
how about to set timeout like this:
`my $timeout = get_var(MAX_JOB_TIME) || value<7200 `

Reply:
The module will be called by a bunch of tests, each has difference time costing.
180 * 60 is to rewrite the default 7200, we have confirmed it's capability for our tests (in qa_run.pm), this time setup in here won't hurt openQA performance, and I want to avoid add too many 'var' configurations in test set :)

Reply:
Checked test list, few need specify timeout, use MAX_JOB_TIME configuration is accepted :)
Done!